daemon/config.go

```diff
@@ -218,6 +244,28 @@ func getConflictFreeConfiguration(configFile string, flags *flag.FlagSet) (*Conf
 		return nil, err
 	}
 
+	// Sanitize values of null so that configurations like
+	// `{ "max-downloads" : null }` will not be considered as 0.
+	// Note: Only the first level is checked.
+	var jsonData map[string]interface{}
+	if err := json.Unmarshal(b, &jsonData); err == nil {
+		// Locate all fields with null value
+		var keys []string
+		for key, value := range jsonData {
+			if value == nil {
+				keys = append(keys, key)
+			}
+		}
+		// Remove fields with null value
+		for _, key := range keys {
+			delete(jsonData, key)
+		}
+		// Update the data
+		if jsonBytes, err := json.Marshal(jsonData); err == nil {
+			b = jsonBytes
+		}
+	}
```

Comment:
You shouldn't need this extra `Unmarshal`/`Marshal`. It should be enough to check if `MaxDownloads` or `MaxUploads` is `nil` after unmarshalling, now that they are pointers.